src/components/APIInterface.js

```diff
@@ -0,0 +1,88 @@
+import { fetchAndParse } from "../fetchAndParse";
+
+export class APIInterface {
+    constructor(apiUrl, cancelSignal) {
```

Comment:
The purpose of cancelSignal was to allow the component to cancel a fetch request if the component unmounts. (This was done to prevent the tests from giving warnings in the logs.) 

The `cancelSignal` was needed in the fetch call's argument, and the `fetchCanceler` was called by the component in `componentWillUnmount` See: https://julietonyekaoha.medium.com/react-cancel-all-axios-request-in-componentwillunmount-e5b2c978c071 or https://gist.github.com/aderbas/187c3ed23db7d6ece9748a5858a8f0d5

Since there will now be an object associated with the fetch calls, you could probably remove the `cancelSignal` from APIInterface's constructor argument and just create `fetchCanceler` and `cancelSignal` inside the constructor of APIInterface. Then have APIInterface expose an `abort()` method for `componentWillUnmount()` to call. This would simplify the function signature to one argument.

As for the need to cancel individual requests, I don't think that's necessary.  If the whole component is unmounted, all requests are cancelled.

Reply:
If it's decided to implement one global APIInterface as per
https://github.com/vgteam/sequenceTubeMap/pull/381#discussion_r1428531987 , then you could keep the `fetchCanceler` and the `cancelSignal` in the component and pass the cancelSignal into each of the APIInterface methods as arguments. (It's OK to pass null or undefined if you don't need to cancel it.) 

You'd only really need to cancel if you are fetching in `componentDidMount` and want to avoid warnings if the component unmounts before the fetch call returns.